nomos-da/network/core/src/protocols/dispersal/executor/behaviour.rs

```diff
@@ -61,31 +61,33 @@ pub enum DispersalError {
 }
 
 impl DispersalError {
+    #[must_use]
     pub const fn blob_id(&self) -> Option<BlobId> {
         match self {
-            Self::Io { blob_id, .. } => Some(*blob_id),
-            Self::Serialization { blob_id, .. } => Some(*blob_id),
-            Self::Protocol {
+            Self::Io { blob_id, .. }
+            | Self::Serialization { blob_id, .. }
+            | Self::Protocol {
```

Comment:
what lint is this? How do they decide when is more readable one than the other version? :thinking:

Reply:
This is a `clippy::match_same_arms`. Again, I think it highlights better the fact that all those variants should do the same operation, and it's easier to see. Otherwise you'd have to compare each variant to understand whether they all do the same thing.

Reply:
Yeah, it makes sense why. But you could argue against it as well. But if it is the consensus, then ahead.